packages/storage/amplify_storage_s3_dart/lib/src/model/s3_list_options.dart

```diff
@@ -56,6 +57,32 @@ class S3ListOptions extends StorageListOptions {
           excludeSubPaths: excludeSubPaths,
         );
 
+  /// {@macro storage.amplify_storage_s3.list_options}
+  ///
+  /// Use this when list all objects without pagination.
+  const S3ListOptions.listAll({
```

Comment:
Is listAll the primary consideration for a list operation? Just wondering why these are factories instead of options?

Reply:
Alternative considered:
```dart
const S3ListOptions({
  StorageAccessLevel accessLevel = StorageAccessLevel.guest,
  int pageSize = 1000,
  bool excludeSubPaths = false,
  bool listAll = false,
  String? nextToken,
})
```
Was thinking `pageSize` and `listAll` are both params of the options constructor could be confusing, as if `listAll` is set to `true`, `pageSize` becomes ineffective. Open for suggestions :)

Reply:
Ah okay that makes sense then 👍